### PR TITLE
feat: cuda node

### PR DESCRIPTION
**Pull request by martabal**
To do : 
- [x] Download cudnn
- [ ] Create arm64 version

### CODE REVIEW

**Comment by hydazz** [comment]
oi

**Comment by martabal** [comment]
We can download cudnn from [this repo](https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/), and 
[here](https://onnxruntime.ai/docs/execution-providers/CUDA-ExecutionProvider.html) is the list of the packages we need to install

**Comment by hydazz** [comment]
> We can download cudnn from [this repo](https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/), and [here](https://onnxruntime.ai/docs/execution-providers/CUDA-ExecutionProvider.html) is the list of the packages we need to install

why do we need these packages when the main image works without them?

**Comment by martabal** [comment]
Only clip encode and tag objects work, not face recognition

File: Dockerfile [hunk 1]
```diff
@@ -35,27 +30,41 @@ RUN \
...
     /tmp/immich --strip-components=1 && \
   echo "**** build machine-learning ****" && \
-  cd /tmp/immich/machine-learning && \
-  pip install --break-system-packages -U --no-cache-dir --pre -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html \
-    coloredlogs \
-    flatbuffers \
+  cd /tmp/immich/machine-learning && \ 
+  python3 -m venv /lsiopy/venv && \
+  /lsiopy/venv/bin/pip install -U --no-cache-dir --extra-index-url https://download.pytorch.org/whl/cu117 \
+    torch==1.13.1 \
```
Comment (hydazz) [inline Dockerfile]: pin this to `${TORCH_VERSION}`

File: Dockerfile [hunk 2]
```diff
@@ -1,11 +1,12 @@
 # syntax=docker/dockerfile:1
 
-FROM ghcr.io/imagegenius/baseimage-ubuntu:lunar
+FROM ghcr.io/imagegenius/baseimage-ubuntu:jammy
 
 # set version label
 ARG BUILD_DATE
 ARG VERSION
 ARG TORCH_VERSION
+ARG TORCHVISION_VERSION
```
Comment (hydazz) [inline Dockerfile]: this is not going to be defined during build...

Comment (martabal) [reply Dockerfile]: So we let the hardcoded version ?

Comment (hydazz) [reply Dockerfile]: well `TORCH_VERSION` is going to be passed through from build so might as well pin it to `${TORCH_VERSION}`, torch vision can be hard pinned in the dockerfile

**Comment by hydazz** [comment]
was going to install large packages to /config without venv... https://github.com/imagegenius/docker-immich-cuda-node/blob/main/root/etc/s6-overlay/s6-rc.d/init-config-ml/run

**Comment by martabal** [comment]
Oh yeah we can totally reduce the image size with this.

**Comment by hydazz** [comment]
can you redo with the changes in this commit https://github.com/imagegenius/docker-immich-cuda-node/commit/9d0bb20b23990050980fbc3ab1c8bdd04a59c950 :)

**Comment by martabal** [comment]
Ubuntu jammy (22.04) uses python 3.10 and `--break-system-packages` is an argument for python 3.11

**Comment by hydazz** [comment]
> Ubuntu jammy (22.04) uses python 3.10 and `--break-system-packages` is an argument for python 3.11

why dont we use lunar?

**Comment by hydazz** [comment]
@martabal if the image is >1gb compressed, then well just package everything into the image, if its getting over 1gb then we'll put torch in appdata

**Comment by hydazz** [comment]
ready @martabal?

**Comment by martabal** [comment]
Not yet, I need to make some changes

**Comment by hydazz** [comment]
@martabal ive merged this so updates will use immich, fork and create a new branch on the fork, if you create a branch here itll build every commit pushed to it :)

**Pull request merged**